samples/buildstrategy/ko/buildstrategy_ko_cr.yaml

```diff
@@ -0,0 +1,83 @@
+---
+apiVersion: build.dev/v1alpha1
+kind: ClusterBuildStrategy
+metadata:
+  name: ko
+spec:
+  buildSteps:
+    - name: prepare
```

Comment:
Is that possible to use the single `build-and-push` step? Or do we some permission problem when building by `ko` with root?

Reply:
One can probably run as root, I was just keeping it similar to the Buildpacks sample where we run the "real" logic as non-root.

Reply:
Can we keep it similar to the Kaniko?

I think it is better to keep it simple as possible.

Reply:
@zhangtbj I think is fine to have it like this. The more strategies we have that can run non-root the better. Otherwise users will be constrained to their infra. Similar to buildah in IKS.